etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/ResultSetIterator.java

```diff
@@ -0,0 +1,41 @@
+package gov.hhs.cdc.trustedintermediary.external.database;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * ResultSetIterator iterates over a SQL {@link ResultSet}. This helps stream over a {@link
+ * ResultSet}.
+ */
+public class ResultSetIterator implements Iterator<ResultSet> {
```

Comment:
Thoughts on implementing the `AutoCloseable` interface to manage the closure of `ResultSet` in case the iterator is not fully traversed until the end?

Reply:
That is handled outside of the iterator in `PostgresDao#fetchData` and `PostgresDao#fetchManyData` already.  We put the `ResultSet` inside the try-with-resources immediately as we get the `ResultSet`.

Reply:
I see it.